src/Features/Core/Portable/Diagnostics/BuildOnlyDiagnosticsService.cs

```diff
@@ -12,22 +12,17 @@
 namespace Microsoft.CodeAnalysis.Diagnostics;
 
 [ExportWorkspaceServiceFactory(typeof(IBuildOnlyDiagnosticsService), ServiceLayer.Default), Shared]
-internal sealed class BuildOnlyDiagnosticsServiceFactory : IWorkspaceServiceFactory
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class BuildOnlyDiagnosticsServiceFactory() : IWorkspaceServiceFactory
 {
-    [ImportingConstructor]
-    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-    public BuildOnlyDiagnosticsServiceFactory()
-    {
-    }
-
     public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
         => new BuildOnlyDiagnosticsService(workspaceServices.Workspace);
 
     private sealed class BuildOnlyDiagnosticsService : IBuildOnlyDiagnosticsService
     {
         private readonly object _gate = new();
         private readonly Dictionary<DocumentId, ImmutableArray<DiagnosticData>> _documentDiagnostics = [];
-        private readonly Dictionary<ProjectId, ImmutableArray<DiagnosticData>> _projectDiagnostics = [];
```

Comment:
data in this was never read in any product code.  only in a single test.  remove it, and updated codepaths affected by it.